docs/samples/tests/xunit/DisposeComponentsTest.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using Microsoft.AspNetCore.Components;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using Xunit;
+
+namespace Bunit.Docs.Samples;
+
+public class DisposeComponentsTest
+{
+  [Fact]
+  public void DisposeElements()
+  {
+    using var ctx = new TestContext();
+    var trackingService = new Mock<ITrackingService>();
+    ctx.Services.AddScoped<ITrackingService>(_ => trackingService.Object);
+    var cut = ctx.RenderComponent<DisposableComponent>();
+    ctx.DisposeComponents();
+
+    ctx.Services.GetRequiredService<NavigationManager>().NavigateTo("newurl");
+
+    trackingService.Verify(t=> t.SetTouchPoint(It.IsAny<string>()), Times.Never);
```

Comment:
```suggestion
    using var ctx = new TestContext();
    var calledTimes = 0;
    var cut = ctx.RenderComponent<DisposableComponent>(parameters => parameters
      .Add(p => p.LocationChangedCallback, url => calledTimes++)
    );
    
    ctx.DisposeComponents();

    ctx.Services.GetRequiredService<NavigationManager>().NavigateTo("newurl");

    Assert.Equal(0, calledTimes);
```

What do you think about this?